### PR TITLE
Keep loaded OpenAPI documents

### PR DESCRIPTION
**Pull request by k2tzumi**
refs https://zenn.dev/katzumi/scraps/f00a2d3e177b77#comment-24254b064a4eb5

### CODE REVIEW

File: http_validator.go [hunk 1]
```diff
@@ -25,6 +25,12 @@ type UnsupportedError struct {
 	Cause error
 }
 
+var oasLoader *openapi3.Loader
+
+func init() {
+	oasLoader = openapi3.NewLoader()
+}
```
Comment (k2tzumi) [inline http_validator.go:32]: How can we implement singleton in golang?
I'm thinking that this process doesn't have to be that thread-safe.

**Review by k2tzumi — COMMENTED** [review]
@k1LoW 
https://zenn.dev/katzumi/scraps/f00a2d3e177b77#comment-bc7c8680c8430d

Coverage commands also benefited

File: http_validator.go [hunk 2]
```diff
@@ -63,31 +69,26 @@ type openApi3Validator struct {
...
 				return nil, err
 			}
 		default:
-			b, err := readFile(l)
-			if err != nil {
-				return nil, err
-			}
-			doc, err = loader.LoadFromData(b)
+			var err error
+			doc, err = oasLoader.LoadFromFile(l)
```
Comment (k2tzumi) [inline http_validator.go:85]: `LoadFromData` was not cached.

**Comment by k1LoW** [comment]
@k2tzumi 
GREAT WORK!!!!!!!!!!!!!! 🚀 🚀 🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

I am going to review it.

File: http_validator.go (hunk 1 above)
Comment (k1LoW) [reply http_validator.go:32]: NewLoader() may be executed only when openapi3.Loader is needed (e.g. sync.Once).

Comment (k1LoW) [reply http_validator.go:32]: However, the performance is good enough to be better than it is now, so I think it is fine as it is.
(We can improve it another time.)

**Review by k1LoW — APPROVED** [review]
GREAT!

**Pull request merged**